### PR TITLE
add reconciler and output templates, reorganize a little

### PR DESCRIPTION
**Pull request by ilackarms**
BOT NOTES: 
resolves https://github.com/solo-io/skv2/issues/78
resolves https://github.com/solo-io/skv2/issues/77

### CODE REVIEW

File: contrib/codegen/funcs/funcs.go [hunk 1]
```diff
@@ -80,37 +80,43 @@ func clientImportPath(grp importedGroup) string {
...
+			selectedResources = append(selectedResources, importedGroup{
+				Group:    filteredGroup,
+				GoModule: clientModule,
+			})
+		}
 	}
 
+	sort.SliceStable(selectedResources, func(i, j int) bool {
+		return selectedResources[i].GoModule < selectedResources[j].GoModule
+	})
```
Comment (EItanya) [inline contrib/codegen/funcs/funcs.go:119]: Should we use the `GVK.String()` here instead? Go module isn't guaranteed to be different.

Comment (ilackarms) [reply contrib/codegen/funcs/funcs.go:119]: resources are randomized across their gomodule, so sorting by gomodule should be sufficient

File: contrib/codegen/templates/input/input_reconciler.gotmpl [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+{{- $kindLowerCamelPlural := pluralize $kindLowerCamel }}
+    {{ $controller_import_prefix }}.Multicluster{{ $resource.Kind }}Reconciler
+{{- end }}
+{{- end }}
+}
+var _ multiClusterReconciler = &multiClusterReconcilerImpl{}
+
+type multiClusterReconcilerImpl struct {
+    ctx       context.Context
+    reconcileFunc func() error
```
Comment (EItanya) [inline contrib/codegen/templates/input/input_reconciler.gotmpl:49]: Can this function take a `clusterName`?

File: contrib/custom_top_level_templates.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
+package contrib
+
+import (
+	"github.com/solo-io/skv2/codegen/model"
+	"github.com/solo-io/skv2/contrib/codegen/funcs"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// use to get all cross-group templates in contrib
+func AllCrossGroupTemplates(params CrossGroupTemplateParameters) []model.CustomTemplates {
```
Comment (EItanya) [inline contrib/custom_top_level_templates.go:10]: As discussed on slack, this won't work as the `OutputFilename` will match for all 3 and overwrite

File: contrib/codegen/templates/input/input_snapshot.gotmpl [hunk 4]
```diff
@@ -1,8 +1,22 @@
-// Definitions for Input Snapshots
-package input
```
Comment (EItanya) [inline contrib/codegen/templates/input/input_snapshot.gotmpl:2]: Why'd you remove the package?

File: contrib/codegen/templates/input/input_reconciler.gotmpl [hunk 5]
```diff
@@ -0,0 +1,151 @@
+{{ $groups := imported_groups }}
```
Comment (EItanya) [inline contrib/codegen/templates/input/input_reconciler.gotmpl:1]: why no package name?

**Pull request merged**